src/modm/architecture/interface/can_filter.cpp

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2020, Raphael Lehmann
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <modm/architecture/interface/can_filter.hpp>
+
+namespace modm::can
+{
+
+modm::IOStream&
+operator<<(modm::IOStream& s, const modm::can::Filter f)
+{
+	if(f.mode == Filter::Mode::ClassicMask) {
+		s.printf("id=%04lx,mask=%04lx", f.identifier, f.mask);
+	}
+	else { // if(f.mode == Filter::Mode::DualIdentifier) 
+		s.printf("id=%04lx,id2=%04lx", f.identifier, f.mask);
```

Comment:
Could you convert this to << syntax? The whole printf is quite large and it is fully optional.
```suggestion
		s << "id=" << modm::hex << uint32_t(f.identifier) << modm::ascii << ",id2=" << modm::hex << uint32_t(f.mask) << modm::ascii;
```